hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/ValidatorRegistry.java

```diff
@@ -46,6 +52,9 @@ public class ValidatorRegistry {
   private final EnumMap<ValidationCondition,
       EnumMap<Type, EnumMap<RequestProcessingPhase, List<Method>>>>
       validators = new EnumMap<>(ValidationCondition.class);
+  private final Map<Pair<Type, RequestProcessingPhase>, Pair<List<Method>, TreeMap<Integer, Integer>>>
+      maxAllowedVersionValidatorMap = new HashMap<>(Type.values().length * RequestProcessingPhase.values().length,
+      1.0f);
```

Comment:
If we get rid `ValidationCondition` then we just have 3 keys to identify the validator(s) that need to run:
- Request type
- Request phase
- Client version

A triple nested map of these keys will be much easier to follow than the wild data structure created here.

Reply:
What do we do for the CLUSTER_NEEDS_FINALIZATION ?